### ISSUE TITLE
[java] BrokenNullCheck: false positive within ternary operator

### ISSUE THREAD

**Issue by Noah0120** [pmd/pmd#3430]
**Affects PMD Version:** 
6.36.0

**Rule:** [BrokenNullCheck](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#brokennullcheck)

**Description:**
A false positive is generated 
when the null checked variable is in the `then` branch of a `conditional expression`.
I think it may not throw NPE.

**Code Sample demonstrating the issue:**

```java
import java.util.ArrayList;
import java.util.List;

class BrokenNullCheck {
    private List<Object> properties = new ArrayList<>();

    void test(int index, Object replacement, Object old) {
        if (index < 0 || (replacement != null ? properties.set(index, replacement) : properties.remove(index)) != old) { // detected
            // ...
        }
    }
}
```

**Expected outcome:**
PMD should not generate any alarms.
It may be a false positive.

Thank you for your consideration.

**Running PMD through:** *[CLI]*

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
